chevah/python-distribution
# [#2801] Create new python distribution.

# Scope

This is a new way of creating and distributing the python package.

To not break the current build system a new repo was created. 

The python distribution will only contain python + pip + setuptools. This should be enough to install other packages like pyopenss/pysqlite... etc

pyopenssl/pysqlite will be built from separate repos, each module will have its own repo and own version.
# Changes

Buildmaster was updated to include a builder for the new python-distribution http://build.chevah.com/waterfall?tag=python-distribution  ...for now it has Ubuntu and Win XP.
We might want to replace Ubuntu with Linux generic ... but I want to see how the linux generic is handled in python-package and brink.

The new paver.sh should just take care of downloading the python distribution and then invoke a few hook based on paver.conf

The new build system re-user part of the old build system just that it no longer keep all source code in the repo and does not build pysqlite/pyopenssl/pycryto...etc

I don't know what is the best way to patch the upstream version... for now, chevahbs has a separate 'patch' step which can be used to execute command which modified the upstream version.

It no longer has production/testing as each build has a specific version.

It requires a modified paver.sh and will also require a modified brink

DEFAULT_VALUE file is still created as I don't know how to call the script back from python.

montage.sh detect_os to show those values to stdout without writing the file

paver.sh and paver.conf were renamed to montage.sh and montage.conf as they no longer depend on paver.

montage.sh can by used with any other build system... including python setup.py.
All brink specific code and chevah custom build system code is included in the montage.conf file. 

The python-distro is now the source for montage.sh

The new pip version automatically caches the files so we no longer need to cache... the cache is still used for python binaries... but in the future we might want to move this to per user cache, rather than per repo

A new brink version was created to support the new build system https://github.com/chevah/brink/pull/70
# How to test

reviewers: @dumol 

check that the new build system make sense and is well documented.

`./venv.sh`
`./venv.sh virtualenv PATH-TO-NEW-VENV`
`./venv.sh clean`

First remote build was successful http://build.chevah.com/waterfall?tag=python-distribution

There are still many things to be fixed but I have created separate tickets for them https://github.com/chevah/python-distribution/issues

Thanks!
